doc/howtos/rdtraining/compute_onchange.rst

```diff
@@ -0,0 +1,295 @@
+:class:`~odoo.fields.Many2many` or a :class:`~odoo.fields.One2many`. An example can be found
+`here <https://github.com/odoo/odoo/blob/713dd3777ca0ce9d121d5162a3d63de3237509f4/addons/account/models/account_reconcile_model.py#L248-L251>`__.
+
+Let's try it on our module with the following exercise!
+
+.. exercise:: Compute the best offer
+
+    - Add the ``best_price`` field to ``estate.property``. It is defined as the maximum of the
+      offers' ``price``.
+
+    - Add the field in the form view as depicted on the first image of the **Goal**.
+
+    Tip: you might give a try to the :meth:`~odoo.models.BaseModel.mapped` method. See
+    `here <https://github.com/odoo/odoo/blob/f011c9aacf3a3010c436d4e4f408cd9ae265de1b/addons/account/models/account_payment.py#L686>`__
+    for a simple example.
+
+Inverse Function
+----------------
+
+You might have noticed that computed fields are read-only by default. This is expected since the
+user is not supposed to set any value.
+
+In some cases, it might be useful to be able to set a value directly. In our real estate example,
+we can define a validity duration for an offer, and set a validity date. We would like to be able
+to set either the duration or the date, one impacting the other.
+
+In this case Odoo provides the ability to use an ``inverse`` function::
+
+    from odoo import api, fields, models
+
+    class TestComputed(models.Model):
+        _name = "test.computed"
+
+        total = fields.Float(compute="_compute_total", inverse="_inverse_total")
+        amount = fields.Float()
+
+        @api.depends("amount")
+        def _compute_total(self):
+            for record in self:
+                record.total = 2.0 * record.amount
+        
+        def _inverse_total(self):
+            for record in self:
+                record.amount = record.total / 2.0
+
+An example can be found
+`here <https://github.com/odoo/odoo/blob/2ccf0bd0dcb2e232ee894f07f24fdc26c51835f7/addons/crm/models/crm_lead.py#L308-L317>`__.
+
+.. exercise:: Compute a validity date for offers
+
+    - Add the following fields to the ``estate.property.offer`` model:
+
+    ========================= ========================= =========================
+    Field                     Type                      Default
+    ========================= ========================= =========================
+    validity                  Integer                   7
+    date_deadline             Date
+    ========================= ========================= =========================
+
+    The ``date_deadline`` is a computed field defined as the offer ``create_date`` to which the
+    ``validity`` is added. Define the appropriate inverse function so that the user can set the
+    date or the validity.
+
+    Tip: the ``create_date`` is only filled in when the record is created. At creation, you will
+    need a fallback to prevent crashing.
+
+    - Add the fields in the form and list view as depicted on the second image of the **Goal**.
```

Comment:
Even if it's shown in the video, I think it's easy to miss the fact that we have to save the property in order for the change to be applied. It's quite different from the behavior of `compute`.

Reply:
note added